pkg/scheduler/plugins/multihierarchyelasticquota/quota_info.go

```diff
@@ -0,0 +1,184 @@
+	Type                             QuotaType          `json:"type"`
+	RuntimeVersion                   int64              `json:"runtimeVersion,omitempty"`
+	ResourceVersion                  string             `json:"resourceVersion,omitempty"`
+	NeedScaleMinQuotaWhenOverRootRes bool               `json:"needScaleMinQuotaWhenOverRootRes,omitempty"`
+	AllowLentResource                bool               `json:"min_quota_share"`
+	CalculateInfo                    QuotaCalculateInfo `json:"calculateInfo,omitempty"`
+	lock                             sync.Mutex
+}
+
+func NewQuotaInfo(isParent, allowLentResource bool, name, parentName QuotaNameType, quotaType QuotaType) *QuotaInfo {
+	return &QuotaInfo{
+		Name:              name,
+		ParentName:        parentName,
+		IsParent:          isParent,
+		AllowLentResource: allowLentResource,
+		Type:              quotaType,
+		RuntimeVersion:    0,
+		CalculateInfo: QuotaCalculateInfo{
+			Max:                       util.NewResourceList(),
+			CalMin:                    util.NewResourceList(),
+			OriMin:                    util.NewResourceList(),
+			Used:                      util.NewResourceList(),
+			Request:                   util.NewResourceList(),
+			SharedRatio:               util.NewResourceList(),
+			Runtime:                   util.NewResourceList(),
+			SubGroupsTotalResLastTime: util.NewResourceList(),
+		},
+	}
+}
+
+func (qi *QuotaInfo) DeepCopy() *QuotaInfo {
+	qi.lock.Lock()
+	defer qi.lock.Unlock()
+
+	return &QuotaInfo{
+		Name:                             qi.Name,
+		ParentName:                       qi.ParentName,
+		IsParent:                         qi.IsParent,
+		AllowLentResource:                qi.AllowLentResource,
+		Type:                             qi.Type,
+		RuntimeVersion:                   qi.RuntimeVersion,
+		NeedScaleMinQuotaWhenOverRootRes: qi.NeedScaleMinQuotaWhenOverRootRes,
+		CalculateInfo: QuotaCalculateInfo{
+			Max:                       qi.CalculateInfo.Max.DeepCopy(),
+			CalMin:                    qi.CalculateInfo.CalMin.DeepCopy(),
+			OriMin:                    qi.CalculateInfo.OriMin.DeepCopy(),
+			Used:                      qi.CalculateInfo.Used.DeepCopy(),
+			Request:                   qi.CalculateInfo.Request.DeepCopy(),
+			SharedRatio:               qi.CalculateInfo.SharedRatio.DeepCopy(),
+			Runtime:                   qi.CalculateInfo.Runtime.DeepCopy(),
+			SubGroupsTotalResLastTime: qi.CalculateInfo.SubGroupsTotalResLastTime.DeepCopy(),
+		},
+	}
+}
+
+func (qi *QuotaInfo) GetLimitRequest() v1.ResourceList {
+	limitRequest := util.NewResourceList()
+	for resName, quantity := range qi.CalculateInfo.Request {
+		if maxQuantity, ok := qi.CalculateInfo.Max[resName]; ok {
+			if quantity.Cmp(maxQuantity) == -1 {
+				//req < Max
+				limitRequest[resName] = quantity.DeepCopy()
+			} else {
+				limitRequest[resName] = maxQuantity.DeepCopy()
+			}
+		}
+	}
+	return limitRequest
+}
+
+func (qi *QuotaInfo) GetIsParent() bool {
```

Comment:
please unify func naming rule. if need lock,  please like LockGetRequest;  by the way, bool seems no need lock